### PR TITLE
XB stability update

### PR DESCRIPTION
**Pull request by jvega190**
- Retrieve parent model (recursive) when the current model is embedded to lock proper item
- Update all matching elements of an updated field instead of only first match
- Fetch level-descriptors

### CODE REVIEW

File: ui/guest/src/store/util.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export function beforeWrite$<T extends any = 'continue', S extends any = never>(
           tap(({ payload }) => payload.type !== 'continue' && post(unlockItem({ path }))),
           switchMap(({ payload }) => (payload.type === 'continue' ? continue$ : stop$))
         );
-      } else if (item.commitId !== localItem.commitId && item.lockOwner !== username) {
+      } else if (item.commitId !== localItem?.commitId && item.lockOwner !== username) {
```
Comment (rart) [inline ui/guest/src/store/util.ts]: Do we actually want an out of sync if the local item is null? Doesn't sound right.

File: ui/guest/src/react/GuestProxy.tsx [hunk 2]
```diff
@@ -419,9 +419,9 @@ export function GuestProxy() {
           if (fieldType === 'image') {
             const tagName = updatedField.eq(index).prop('tagName').toLowerCase();
             if (tagName === 'img') {
-              updatedField.eq(index).attr('src', value);
```
Comment (rart) [inline ui/guest/src/react/GuestProxy.tsx:422]: What was going on here?
Please retest this in other places too.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/react/GuestProxy.tsx (hunk 2 above)
Comment (jvega190) [reply ui/guest/src/react/GuestProxy.tsx:422]: updatedField is a results of a query of all elements matching a modelId+fieldId, and it was only updating the src of one element (at `index`). So if we have one component in multiple parts (for example wordify, in the slider and in the latest content cards) it won't update all of the instances

Comment (rart) [reply ui/guest/src/react/GuestProxy.tsx:422]: The index needs to be **considered** for repeat-group-nested fields/item.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/react/GuestProxy.tsx [hunk 3]
```diff
@@ -408,24 +408,31 @@ export function GuestProxy() {
...
+              updatedField.attr('src', value);
             } else {
-              updatedField.eq(index).css('background-image', `url(${value})`);
+              updatedField.css('background-image', `url(${value})`);
             }
           } else if (fieldType === 'video-picker') {
-            updatedField.eq(index).find('source').attr('src', value);
-            updatedField.eq(index)[0].load();
+            updatedField.find('source').attr('src', value);
+            $(updatedField).each((index, element) => {
```
Comment (rart) [inline ui/guest/src/react/GuestProxy.tsx]:
```suggestion
            updatedField.each((index, element) => {
```

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**